src/main/java/com/mineboundteam/minebound/magic/MagicEvents.java

```diff
@@ -128,34 +127,34 @@ protected static void handlePlayerArmor(Player player) {
         };
 
         for (ItemStack item : player.getArmorSlots()) {
-            if (!item.isEmpty())
-                item.getCapability(ArmorRecoveryProvider.ARMOR_RECOVERY).ifPresent(recovery -> {
-                    MyrialArmorItem armorItem = (MyrialArmorItem) item.getItem();
-                    if (armorItem.getDamage(item) >= armorItem.getMaxDamage(item))
-                        recovery.recovering = true;
-                    if (1.0d * armorItem.getDamage(item) / armorItem.getMaxDamage(item) <= 0.25d)
-                        recovery.recovering = false;
-
-                    if (tier.max == null || tier.max.getValue() < armorItem.getTier().getValue())
-                        tier.max = armorItem.getTier();
-
-                    int armorDamage = armorItem.getDamage(item);
-                    if (armorDamage > 0) {
-                        FoodData pFoodData = player.getFoodData();
-                        if (pFoodData.getSaturationLevel() > 0 && pFoodData.getFoodLevel() >= 20) {
-                            float f = Math.min(pFoodData.getSaturationLevel(), 6.0F);
-                            armorItem.setDamage(item, (int) (armorDamage - f / 6));
-                            pFoodData.addExhaustion(f / 16.0f);
-                        } else if (pFoodData.getFoodLevel() >= 16) {
-                            armorItem.setDamage(item, armorDamage - 1);
-                            pFoodData.addExhaustion(0.125f);
-                        }
-
-                        if (armorItem.getDamage(item) < 0)
-                            armorItem.setDamage(item, 0);
+            if (!item.isEmpty() && item.getItem() instanceof MyrialArmorItem armorItem){
+                boolean recovering = item.getTag().getBoolean("recovery");
```

Comment:
Should use `getOrCreateTag()` instead of `getTag()` everywhere just to play it safe.

Reply:
getOrCreateTag creates a tag within the overall item CompoundTag, I believe; getTag() retrieves the ItemStack's base tag, or creates a new CompoundTag if it doesn't exist. I can check to make sure of that when I get out of class, but I'm pretty sure none of the tag functions will ever return null

Reply:
`getOrCreateTag()` does what you think `getTag()` does. `getTag()` just returns the base tag which might be null.

Reply:
Here is that code from `ItemStack.java`
```java
@Nullable
public CompoundTag getTag() {
   return this.tag;
}
public CompoundTag getOrCreateTag() {
   if (this.tag == null) {
      this.setTag(new CompoundTag());
   }
   return this.tag;
}
```

Reply:
Alright, should be fixed; I'll test later

Reply:
There's one more usage of `getTag()` at line 198.

Reply:
fixed